### ISSUE TITLE
[RFE] Add support for OpenCost in KubePlus

### ISSUE THREAD

**Issue by chiukapoor** [cloud-ark/kubeplus#1287]
### Description:
As KubePlus aims to provide a comprehensive solution for managing multi-instance multi-tenancy (MIMT) applications on Kubernetes, it would be beneficial to integrate support for OpenCost into the project. 

### Overview:
OpenCost is a cost management platform that provides insights and analytics into resource utilization and spending within Kubernetes clusters. By adding support for OpenCost into KubePlus, users will gain visibility into the resource consumption and associated costs of each application instance deployed using KubePlus.

### Proposal:

**Integration:** Implement functionality within KubePlus to capture resource consumption data for each application instance.
**Data Export:** Define an interface for exporting resource consumption data to OpenCost in a compatible format.
**Documentation:** Update KubePlus documentation to include instructions on configuring and using the OpenCost integration.
**Testing:** Conduct thorough testing to ensure the reliability and accuracy of the OpenCost integration.

### Benefits:
**Cost Visibility:** Users will have clear insights into each application instance's resource usage and associated costs.
**Optimization:** With visibility into resource consumption, users can optimize their deployments to minimize costs and maximize efficiency.
**Billing:** Enables billing and chargeback mechanisms based on actual resource usage.

**Comment by devdattakulkarni** [comment]
Do we know what is the data format expected by opencost? 

We have a kubectl plugin (kubectl metrics), which collects per-application resource consumption metrics (CPU, memory, storage, network ingress, network egress). We can enhance this plugin to generate output that is expected by opencost.

**Comment by chiukapoor** [comment]
> Also, can opencost tell us what all Pods are considered when calculating the cost for a given namespace? 

Yes, it considers that the cost for all the pods is calculated for a given namespace.

> rather than telling users to install opencost and then use opencost plugin.

The user will need to install opencost to use the functionality, what we can do is have a script which installs Kubeplus and the script will have a parameter that the user can pass to install opencost as well for the cost calculation functionality.

> I think it will be good to add a kubeplus kubectl plugin

Yes, I am looking for the possibility of wrapping kubecost plugin in kubeplus kubectl plugin so that we can use all the kubecost plugin commands directly inside kubeplus kubectl plugin.

**Comment by devdattakulkarni** [comment]
@chiukapoor Sounds like a good plan.

**Comment by devdattakulkarni** [comment]
@chiukapoor Can you refresh my memory on where we stand on this issue?

**Comment by devdattakulkarni** [comment]
@chiukapoor Some time has passed since we discussed about this issue. I think given that OpenCost installed in a cluster is able to show cost per Namespace by default, there is no need to add any specific functionality in KubePlus for this.
Let me know if you think otherwise.

**Comment by devdattakulkarni** [comment]
Closing this for now per above comment. We can revisit this later if needed.
